Seagate/cortx-manager
# CORTX-30326:Hybrid API framework improvement for all the capacity API's

Signed-off-by: Pranali04796 <pranali.ugale@seagate.com>

# Problem Statement
- JIRA id: [CORTX-30326](https://jts.seagate.com/browse/CORTX-30326)
- Hybrid API framework improvement for all the capacity API's
# Design
-  For Bug, Describe the fix here.
-  For Feature, Post the link for design

# Coding
   Checklist for Author

# Testing 
  Checklist for Author
- [ ] Unit and System Tests are added
- [ ] Test Cases cover Happy Path, Non-Happy Path and Scalability
- [ ] Testing was performed with RPM
- Build Details:
<img width="431" alt="o1" src="https://user-images.githubusercontent.com/68841079/163753906-06a68b8c-9b51-4607-bc73-6770335bec41.PNG">
- Deployment details:
<img width="679" alt="o2" src="https://user-images.githubusercontent.com/68841079/163753943-e7f2516c-a206-460f-ae5c-8f2355dfe3d1.PNG">
- Add API in hybrid api's:

1. Add decorator @csmauth.hybrid
<img width="393" alt="hybrid" src="https://user-images.githubusercontent.com/68841079/163754409-264a57b3-63f4-4369-91b2-abbc2b952d4d.PNG">
2. Add api in hybrid api's list
<img width="532" alt="hybrid api" src="https://user-images.githubusercontent.com/68841079/163754462-56c10d22-9a10-4ade-a119-377e5dd2bda0.PNG">


Test case1: conf key value for hyrbid api is **disable**
<img width="451" alt="o4 conf key value disable" src="https://user-images.githubusercontent.com/68841079/163754043-9a4b6ba9-5966-499c-b63d-874379b2ce36.PNG">
Expected result: It should work without bearer token:
<img width="519" alt="o4 conf key value disable api becomes public" src="https://user-images.githubusercontent.com/68841079/163754100-ca95e84e-d1c9-41b2-8a95-862435552de2.PNG">
Test case1: conf key value for hyrbid api is **enable**
<img width="617" alt="o5 conf key with auth enable" src="https://user-images.githubusercontent.com/68841079/163754219-7ea5c4c6-f145-45c3-9a00-6ff1b4a5c9c0.PNG">
Expected result: It should not work without bearer token:
<img width="526" alt="o5 conf key enable token not provided" src="https://user-images.githubusercontent.com/68841079/163754313-8a5c42e0-2af4-4552-b830-49623ece5fd9.PNG">


# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
- [ ] Interface change (if any) are documented
- [ ] Side effects on other features (deployment/upgrade)
- [ ] Dependencies on other component(s)

# Review 
  Checklist for Author

# Documentation
  Checklist for Author
- [ ] Changes done to WIKI / Confluence page / Quick Start Guide
